languages/multi-language/src/main/java/de/jplag/multilang/MultiLanguageParser.java

```diff
@@ -7,18 +7,28 @@
 import java.util.Optional;
 import java.util.Set;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import de.jplag.Language;
 import de.jplag.ParsingException;
 import de.jplag.Token;
 
 public class MultiLanguageParser {
+    private static final Logger LOG = LoggerFactory.getLogger(MultiLanguageParser.class);
+    private static final String WARNING = "This module only allows parsing of multiple languages. No comparisons will be made between languages";
+    private static final String ERROR_MULTIPLE_PRIORITY_LANGUAGES = "Multiple language modules with priority (%s) have been found for file: %s";
+    private static final String ERROR_MULTIPLE_LANGUAGES = "Multiple language modules (%s) have been found for file: %s";
     private final List<Language> languages;
+    private static boolean hasPrintedWarning;
 
     public MultiLanguageParser(MultiLanguageOptions options) {
         this.languages = options.getLanguages();
+        hasPrintedWarning = false;
     }
 
     public List<Token> parseFiles(Set<File> files, boolean normalize) throws ParsingException {
+        this.printWarning();
         List<Token> results = new ArrayList<>();
         for (File file : files) {
             Optional<Language> language = findLanguageForFile(file);
```

Comment:
We discussed that we can find out at compile time whether the mapping of file extensions to language modules is well-defined. I would argue then that we should set up an actual look up table instead of repeating the same steps for each individual file.